src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js

```diff
@@ -13,6 +14,41 @@ const MISSING_BLOCK_ERROR_MESSAGE = 'The "block" field type needs at least one t
 const BLOCK_PREVIEW_TAG = 'sulu.block_preview';
 
 export default class FieldBlocks extends React.Component<FieldTypeProps<Array<BlockEntry>>> {
+    componentDidUpdate(prevProps: FieldTypeProps<Array<BlockEntry>>) {
+        const {defaultType, onChange, types, value} = this.props;
+        const {types: oldTypes} = prevProps;
+
+        if (!types || !oldTypes) {
+            throw new Error(MISSING_BLOCK_ERROR_MESSAGE);
+        }
+
+        const oldValue = toJS(value);
+        let newValue = oldValue;
+
+        if (value && types !== oldTypes) {
+            if (!defaultType) {
+                throw new Error(
+                    'It is impossible that a block has no defaultType. This should not happen and is likely a bug.'
+                );
+            }
+
+            // set block to default type if type does not longer exist
+            // this could happen for example in a template switch
+            newValue = newValue.map((block) => {
+                if (!types[block.type]) {
+                    return {...block, type: defaultType}
```

Comment:
@danrot need to copy the block else the equals check will fail as newValue blocks and oldValue blocks is the same object